### PR TITLE
inline predicates.ncl into output files

### PR DESCRIPTION
**Pull request by Radvendii**
Currently we rely on the files in `./lib` to be present at a magic path at runtime of the nickel code **generated by json-schema-to-nickel** (see #17). This is a high burden on the user. Ultimately, there should be an idiomatic way of packaging things in nickel, so we can refer to a library that is at some well-known internet path. For now, there is no such nickel idiom.

This PR instead inlines the library into the output of json-schema-to-nickel, thus removing the need for the user to have it at any magic path.

**Problems with this solution:**

The use of `import`s in the nickel library causes a huge headache. Without it, we could simply `import_bytes!("./lib.ncl")` and thereby include the nickel library in the rust binary. Instead, we have to parse the library files and resolve all the imports at build time, concatenate them into a big `RichTerm`, pretty print that back out to a generated build file, `import_bytes!` **that** file, and then re-parse it, only to insert it into a bigger `RichTerm` and pretty-print it again. (of course, this is then re-parsed by nickel when the output of json-schema-to-nickel is actually used)

This works, but it is a lot of complexity. The pretty printer is also somewhat finicky, so going back and forth so many times might cause problems at some point.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            &|subterm: RichTerm, cache| {
+                match_sharedterm! { subterm.term, with {
+                    Term::Import(p) => {
+                        let mut pb: PathBuf = path.into();
+                        pb.set_file_name(p);
+                        helper(pb, cache, parser)
+                    }
+                } else {
+                    Ok(subterm)
+                // XXX: how is the formatting supposed to work for these curly braces??
```
Comment (vkleen) [inline build.rs:55]: I think the double closing brace looks pretty okay. The alternative would probably be to add an extra level of indentation to the entire macro contents.
```suggestion
```

File: build.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+fn inline_imports(path: impl Into<OsString>) -> Result<RichTerm, Box<dyn Error>> {
+    fn helper(
+        path: impl Into<OsString>,
+        cache: &mut Cache,
+        parser: &TermParser,
+    ) -> Result<RichTerm, Box<dyn Error>> {
+        let path = &path.into();
+        let file_id = cache.add_file(path.clone())?;
+        let lexer = Lexer::new(cache.source(file_id));
+        let rt = parser.parse_strict(file_id, lexer).unwrap();
```
Comment (vkleen) [inline build.rs:44]: It's not like it makes a huge difference, but if we try to return a `Result` we might as well use `?` everywhere.
```suggestion
        let rt = parser.parse_strict(file_id, lexer)?;
```

File: build.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    let mut cache = Cache::new(ErrorTolerance::Strict);
+    let parser = TermParser::new();
+    helper(path, &mut cache, &parser)
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let lib = inline_imports("./lib/predicates.ncl")?;
+
+    let out_dir = env::var_os("OUT_DIR").unwrap();
```
Comment (vkleen) [inline build.rs:71]:
```suggestion
    let out_dir = env::var_os("OUT_DIR")?;
```

File: build.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    let comptime_rust_lib = Path::new(&out_dir).join("predicates.ncl");
```
Comment (vkleen) [inline build.rs:72]:
```suggestion
    let comptime_predicates = Path::new(&out_dir).join("predicates.ncl");
```
Maybe? But `comptime_rust_lib` definitely doesn't fit.

**Review by vkleen — APPROVED** [review]

File: build.rs (hunk 1 above)
Comment (Radvendii) [reply build.rs:55]: yeah idk the `with {` as well as the double curly brace look pretty weird to me, but i don't think there's a way to make it look normal (to me) with the syntax we have

**Pull request merged**